AudioKit/Common/Internals/Helpers/ExceptionCatcher.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  ExceptionCatcher.swift
+//  AudioKit
+//
+//  Created by Stéphane Peter, revision history on Github.
+//  Copyright © 2018 AudioKit. All rights reserved.
+//
+
+import Foundation
+
+extension NSException: Error
+{
+    public var domain: String { return "io.audiokit.SwiftExceptionCatcher" }
+    public var code: Int { return 0 }
+}
+
+public func AKTry(_ operation: @escaping (() throws -> Void),
+                  finally: (() -> Void)? = nil) throws
+{
+    var error: NSException?
+    
+    let theTry = {
+        do {
+            try operation()
+        } catch let ex {
+            error = ex as? NSException
```

Comment:
If ex is a Swift error, rather than an exception, it will just be ignored here.  We need to figure out how to rethrow here, or do away with the do-try-catch here and make operation non-throwing.

Reply:
I guess we could try to wrap the error in an NSError.  Re-throwing the original would be ideal though.

Reply:
You're probably right about Swift errors/exceptions, however the main goal of this was to specifically intercept Objective-C exceptions, as these are the ones that go uncaught currently. I only use AKTry() around sections of Cocoa code that should only trigger Obj-C exceptions, though there are probably a few exceptions.

Reply:
I agree that the exception cases will be handled, but swallowing Swift errors might lead to troubles down the road.  Rethrowing would be best, but we should at least print them (if not NSException).  I couldn’t figure out a general solution, I worry we might need to handle these on a case by case basis.

Reply:
Do we really have cases where both Swift and NSExceptions can be raised by the single call, i.e. methods that are marked `throws`? We could restrict this to calls that have been known to unexpectedly raise NSException, like `engine.prepare()` for instance. I would be OK with moving current throwing calls out of the AKTry closure if that covers all our bases.

Reply:
My focus has been on the current methods we have that throw Swift errors, and the goal was basically to intercept any NSException and rethrow them in the same way, so we don't get crashes.

Reply:
AVAudioEngine.start() and some of the throwing AVAudioEngine funcs are suspected to throw exceptions.